client/src/featureform/resources.py

```diff
@@ -1397,5 +1397,27 @@ def config(self):
             "Credentials": self.credentials.config(),
         }
 
+@typechecked
+@dataclass
+class SparkExecutor:
```

Comment:
Should see if @simba-git has any thoughts on what this should be called since right now it'll be 
`register_spark(SparkExecutor())`, which feels weird. The other options are EMRCredentials and DatabricksCredentials, are there spark credentials that someone might need, SparkCredentials? Otherwise something like SparkConfig?